src/encoded/static/components/experiment_series.js

```diff
@@ -85,85 +328,93 @@ class ExperimentSeriesComponent extends React.Component {
         super(props);
         this.state = {
             /** Audit objects keyed by dataset @id. Datasets w/o audits included with empty object */
-            auditsByDataset: {},
+            viewableDatasets: {},
         };
-        this.getDatasetAudits = this.getDatasetAudits.bind(this);
+        this.statusFilteredDatasets = this.statusFilteredDatasets.bind(this);
         this.getViewableDatasets = this.getViewableDatasets.bind(this);
     }
 
     componentDidMount() {
-        this.getDatasetAudits();
+        this.getViewableDatasets();
     }
 
     componentDidUpdate() {
-        const relatedDatasets = this.props.context.related_datasets;
-        if (relatedDatasets.length > 0) {
-            // If any @id in related_datasets isn't in this.state.auditsByDataset, retrieve all
-            // dataset audits. This can happen if the user edits the viewed ExperimentSeries
-            // object.
-            const datasetsInAudits = relatedDatasets.some(dataset => !!this.state.auditsByDataset[dataset['@id']]);
-            if (!datasetsInAudits) {
-                this.getDatasetAudits();
-            }
+        const datasetAtIds = Object.keys(this.statusFilteredDatasets());
+        const updatedDatasetAtIds = Object.keys(this.state.viewableDatasets);
+        const symmDiffAtIds = datasetAtIds.filter(
+            i => !updatedDatasetAtIds.includes(i)
+        ).concat(
+            updatedDatasetAtIds.filter(i => !datasetAtIds.includes(i))
+        );
+        if (symmDiffAtIds && symmDiffAtIds.length > 0) {
```

Comment:
`Array.filter` is guaranteed to give you _something_ — a zero-length array at minimum — so you don’t need to check for `symmDiffAtIds`’s existence, though it really doesn’t make a lot of difference performance-wise of course.

Reply:
If people take efforts to guarantee it, I respect that and should make my code simpler 😊.